examples/BCDR/BackupFromStorageQueue.cs

```diff
@@ -0,0 +1,172 @@
+        public static void Run([TimerTrigger("0 */10 * * * *")]TimerInfo myTimer, ILogger log)
+        {
+            try
+            {
+                log.LogInformation($"C# Timer trigger function executed at: {DateTime.Now}");
+                
+                // Create storage queue client
+                QueueClient queueClient = new QueueClient(Environment.GetEnvironmentVariable("AzureWebJobsStorage"), "eventgridqueue");
+
+                // Peek to see if there are events in the queue
+                while (queueClient.PeekMessages().Value.Length > 0 )
+                {
+                    Response<QueueMessage[]> messages = queueClient.ReceiveMessages(maxMessages: 32);
+
+                    // extract list of key+labels from event data
+                    HashSet<KeyLabel> updatedKeyLabels = ExtractKeyLabelsFromEvents(messages.Value, log);
+
+                    // If there are any valid App Configuration events, update secondary store
+                    if(updatedKeyLabels.Count > 0)
+                    {
+                        BackupKeyValues(updatedKeyLabels, log);
+                    }
+
+                    // Delete this batch of events from storage queue
+                    foreach (QueueMessage message in messages.Value)
+                    {
+                        log.LogInformation($"Deleting message: {message.MessageId}");
+                        queueClient.DeleteMessage(message.MessageId, message.PopReceipt);
+                    }
+                }
+            }
+            catch (RequestFailedException exception)
+            {
+                log.LogError($"Service request failed from the server.\nStatus: {exception.Status}\nError Message: {exception.Message}");
+            }
+            catch (AggregateException exception) when ((exception as AggregateException)?.InnerExceptions?.All(ex => ex is RequestFailedException) ?? false)
+            {
+                log.LogError($"Service request failed from the server with the following exceptions:");
+                foreach (RequestFailedException ex in exception.InnerExceptions)
+                {
+                    log.LogError($"\nStatus: {ex.Status}\nError Message: {ex.Message}");
+                }
+            }
+        }
+    }
+
+    public class KeyLabel
+    {
+        public string Key { get; set; }
+        public string Label { get; set; }
+
+        public KeyLabel(string key, string label)
+        {
+            Key = key;
+            Label = label;
+        }
+    }
+
+    public class KeyLabelComparer : IEqualityComparer<KeyLabel>
+    {
+        public bool Equals(KeyLabel x, KeyLabel y)
+        {
+            return string.Equals(x.Key, y.Key, StringComparison.InvariantCulture) 
+                && string.Equals(x.Label, y.Label, StringComparison.InvariantCulture);
+        }
+
+        public int GetHashCode(KeyLabel obj)
+        {
+            return obj.Key.GetHashCode();
```

Comment:
This will cause items with the same key but different labels to be overwritten in the hashset, right?

Reply:
This shouldn't cause such key-values to be overwritten, but it will lead to collisions and increase the runtime of operations for the hash table. We should be able to use a hash function that uses both key and label fields to optimize it.

Reply:
Using a hash function based on both key and label to reduce collisions.